pymtl3/passes/translator/behavioral/BehavioralTranslatorL1.py

```diff
@@ -75,9 +75,14 @@ def translate_behavioral( s, m ):
 
     for upblk_type in ( 'CombUpblk', 'SeqUpblk' ):
       for blk in upblks[ upblk_type ]:
+        upblk_ir = s.behavioral.rtlir[ m ][ blk ]
         upblk_srcs.append( s.rtlir_tr_upblk_src(
-          blk, s.behavioral.rtlir[ m ][ blk ]
+          blk, upblk_ir
         ) )
+        blk.is_lambda = upblk_ir.is_lambda
```

Comment:
is blk the actual function here? are you attaching those metadata back to the function?

Reply:
that's right

Reply:
hmmmm might not be very good. Is there a way to keep this separate from the function itself?

Reply:
The info of upblks is there as the attributes of the upblk behavioral IR. During translation the translator extracts these attributes and supply them to the backend translation callbacks

Reply:
^ This is implemented in the following commit